crawl-ref/source/cloud.h

```diff
@@ -60,3 +60,100 @@ const cloud_tile_info& cloud_type_tile_info(cloud_type type);
 
 void start_still_winds();
 void end_still_winds();
+
+void manage_cloud_generators();
+
+enum class special_fog_type {
+    normal,
+    chained,
+};
+
+class CloudGenerator
+{
+public:
+    CloudGenerator
+    (coord_def _loc, cloud_type _type,
+        int _walk_dist, int _pow_min,
+        int _pow_max, int _pow_rolls,
+        kill_category _kc, int _size_min,
+        int _size_max, int _spread_rate,
+        int _start_clouds, int _excl_rad,
+        int _size_buildup_amnt, int _size_buildup_time,
+        int _spread_buildup_amnt, int _spread_buildup_time,
+        int _buildup_turns, int _delay_min,
+        int _delay_max, special_fog_type _special,
+        string _marker_id, vector<string> _slaves)
+        :loc(_loc), type(_type), walk_dist(_walk_dist),
+        pow_min(_pow_min), pow_max(_pow_max), pow_rolls(_pow_rolls), kc(_kc),
+        size_min(_size_max), size_max(_size_max), spread_rate(_spread_rate), start_clouds(_start_clouds), exclusion_radius(_excl_rad),
+        size_buildup_amnt(_size_buildup_amnt), size_buildup_time(_size_buildup_time),
+        spread_buildup_amnt(_spread_buildup_amnt), spread_buildup_time(_spread_buildup_time),
+        buildup_turns(_buildup_turns), delay_min(_delay_min),
+        delay_max(_delay_max), special(_special),
+        marker_id(_marker_id), slaves(_slaves)
+    {
```

Comment:
I'm not the greatest programmer but surely there's a better way to do this constructor...

Reply:
I'm open to ideas.